svgnative/src/SVGDocumentImpl.h

```diff
@@ -133,42 +133,99 @@ class SVGDocumentImpl
         ElementType Type() const override { return ElementType::kGroup; }
     };
 
-    struct Graphic : public Element
+    struct Reference : public Element
     {
-        Graphic(GraphicStyleImpl& aGraphicStyle, std::set<std::string>& aClasses, FillStyleImpl& aFillStyle, StrokeStyleImpl& aStrokeStyle,
-            std::shared_ptr<Path> aPath)
+        Reference(GraphicStyleImpl& aGraphicStyle, std::set<std::string>& aClasses, FillStyleImpl& aFillStyle, StrokeStyleImpl& aStrokeStyle,
+            std::string aHref)
             : Element(aGraphicStyle, aClasses)
             , fillStyle{aFillStyle}
             , strokeStyle{aStrokeStyle}
-            , path{std::move(aPath)}
+            , href{std::move(aHref)}
         {
         }
 
         FillStyleImpl fillStyle;
         StrokeStyleImpl strokeStyle;
-        std::shared_ptr<Path> path;
+        std::string href;
 
-        ElementType Type() const override { return ElementType::kGraphic; }
+        ElementType Type() const override { return ElementType::kReference; }
     };
 
-    struct Reference : public Element
+    struct Graphic : public Element
     {
-        Reference(GraphicStyleImpl& aGraphicStyle, std::set<std::string>& aClasses, FillStyleImpl& aFillStyle, StrokeStyleImpl& aStrokeStyle,
-            std::string aHref)
+        Graphic(GraphicStyleImpl& aGraphicStyle, std::set<std::string>& aClasses, FillStyleImpl& aFillStyle, StrokeStyleImpl& aStrokeStyle,
+            std::shared_ptr<Path> aPath)
             : Element(aGraphicStyle, aClasses)
             , fillStyle{aFillStyle}
             , strokeStyle{aStrokeStyle}
-            , href{std::move(aHref)}
+            , path{std::move(aPath)}
         {
         }
 
         FillStyleImpl fillStyle;
         StrokeStyleImpl strokeStyle;
-        std::string href;
-
-        ElementType Type() const override { return ElementType::kReference; }
-    };
+        std::shared_ptr<Path> path;
 
+        ElementType Type() const override { return ElementType::kGraphic; }
+        Graphic& operator= (const Reference& refObj)
+        {
+            if (refObj.fillStyle.hasFill)
+            {
+                this->fillStyle.hasFill       = refObj.fillStyle.hasFill;
```

Comment:
in above example you mentioned , by doing recent changes output is showing 'green' which is desired one. As style object of reference object is taken into account by defining assignment operator , which pass on the fill and stroke properties if any from reference object to object that gets referenced.

i have removed the space aligning.

Reply:
```
 <g id ="q" fill="yellow">
    <rect id="r" width="100" height="100"/>
</g>
<use xlink:href="#r" x="120" />
```

in above example ,with the original code (not including any recent changes) the output is first rect is yellow and second is also yellow. even our expectation is that the first rect will be yellow and the second is black.
we have a stack of fill and stroke styles. the first rect takes fill style from parent . That FillStyle will be passed as a reference to function “ParseGraphicProperties” as it modifies FillStyle if any fill attribute found.currently in this scenario, it did not able to modify because no fill attribute is present. Then “Graphic” element with its FillStyle is pushed in stack. **There is no way to distinguish its parent FillStyle and current one.**

when reference element is pointing to any graphic element, simply graphic element will be drawn as it was previously stored in stack. so “yellow” rect will be drawn. if there is any styling applied to reference object it will simply ignored.

i have also debug the area you pointed in discussion
 auto saveRestore = SaveRestoreHelper{mRenderer, reference.graphicStyle};

it simple saves the graphic style , but fill and stroke properties are not stored anywhere which actually fills the rect.

Reply:
@dirkschulze can you pls share your thoughts on above comment ?